NitroxClient/Debuggers/Drawer/Unity/UnityEventDrawer.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace NitroxClient.Debuggers.Drawer.Unity;
+
+public class UnityEventDrawer : IDrawer
+{
+    private const float LABEL_WIDTH = 250;
+
+    public Type[] ApplicableTypes { get; } = { typeof(UnityEvent), typeof(UnityEvent<bool>) };
+
+    public void Draw(object target)
+    {
+        switch (target)
+        {
+            case UnityEvent unityEvent:
+                DrawUnityEvent(unityEvent);
+                break;
+            case UnityEvent<bool> unityEventBool:
+                DrawUnityEventBool(unityEventBool);
+                break;
+        }
+    }
+
+    public static void DrawUnityEvent(UnityEvent unityEvent, string name = "NoName")
+    {
+        using (new GUILayout.HorizontalScope())
+        {
+            GUILayout.Label(name, NitroxGUILayout.DrawerLabel, GUILayout.Width(LABEL_WIDTH));
+            NitroxGUILayout.Separator();
+            if (GUILayout.Button("Invoke All", GUILayout.Width(100)))
+            {
+                unityEvent.Invoke();
+            }
+        }
+
+        DrawUnityEventBase(unityEvent);
+    }
+
+    public static void DrawUnityEventBool(UnityEvent<bool> unityEvent, string name = "NoName")
+    {
+        using (new GUILayout.HorizontalScope())
+        {
+            GUILayout.Label(name, NitroxGUILayout.DrawerLabel, GUILayout.Width(LABEL_WIDTH));
+            NitroxGUILayout.Separator();
+            if (GUILayout.Button("Invoke All (true)", GUILayout.Width(100)))
+            {
+                unityEvent.Invoke(true);
+            }
+            if (GUILayout.Button("Invoke All (false)", GUILayout.Width(100)))
+            {
+                unityEvent.Invoke(false);
+            }
```

Comment:
The button is drawn inside the if statements so to not have it blink away the if without else is needed.

Reply:
Sorry I don't think I made myself clear. What I meant to say was I assume that you can't press both these buttons true and false at once therefore meaning only one of these checks will ever be true when the evaluation is tested. Therefore having a if else would only check the other evaluation if the first isn't run and not every time

Reply:
Yeah but than the other button will be away for that frame bc it will not get rendered as the code that renders both buttons it is inside each if condition.